app.rb

```diff
@@ -0,0 +1,111 @@
+require_relative 'person'
+require_relative 'book'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+
+class App
+  def initialize
+    @book_list = []
+    @people = []
+    @rentals = []
+    @students = []
+    @teachers = []
+  end
+
+  def run_selection
+    display_selection
+  end
+
+  def list_all_books
+    if @book_list.empty?
+      puts 'There is no book in the Library'
+    else
+      @book_list.each_with_index { |book, i| puts "#{i}) Title: #{book.title}, Author: #{book.author}" }
+    end
+  end
+
+  def list_all_people
+    if @people.empty?
+      puts 'There are no people in the Library'
+    else
+      @people.each_with_index do |person, _i|
+        puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = gets.chomp
+    case person_type
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid Options'
+      create_person
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N] '
+    parent_permission = gets.chomp
+    parent_permission = true if parent_permission == 'y'
+    parent_permission = false if parent_permission == 'n'
+    student = Student.new(age, name, parent_permission)
+    @people << student unless @people.include?(student)
+    @students << student unless @students.include?(student)
+    puts 'Student created successfully!'
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(age, specialization, name)
+    @people << teacher unless @people.include?(teacher)
+    @teachers << teacher unless @teachers.include?(teacher)
+    puts 'Teacher created successfully!'
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    puts "Book created successfully!"
+    @book_list << book unless @book_list.include?(book)
+  end
+
+  def create_rental
+    puts "Select a book from the following list by number:\n"
+    list_all_books
+    book_index = gets.chomp.to_i
+    puts "Select a person from the following list by number (not id):\n"
```

Comment:
- [ ] You did a great job by making changes based on the previous reviewer's request. However, while making the changes, you accidentally make another issue. Which is: based on your message in line 94 we need to select the person from the number. But the number is not showing as you described. Please kindly add the number to the given user list based on your message. Thank you!


<img width="577" alt="Screenshot 2023-04-14 at 22 34 14" src="https://user-images.githubusercontent.com/69011963/232117055-dc0b47cb-46da-43a9-84d8-1b8c0cf7cf36.png">

Reply:
Okay, Thanks alot @Rustamxon7, I have implemeted the changes correctly.